src/main/java/Duke.java

```diff
@@ -1,10 +1,104 @@
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke();
+        TaskList tl = new TaskList();
+        String greeting = duke.greetUser();
+        System.out.println(greeting);
+        Scanner sc = new Scanner(System.in);
+        String userInput = sc.nextLine();
+
+        while(!userInput.equals("bye")) {
+            try {
+                if (!userInput.equals("list")) {
+                    if (userInput.startsWith("mark")) {
+                        String[] strArr = userInput.split(" ", 2);
+                        int taskNumber = Integer.parseInt(strArr[1]);
+                        if (taskNumber < 1 || taskNumber > tl.items) {
+                            throw new InvalidTaskException(taskNumber);
+                        } else {
+                            Task taskToMark = tl.list.get(taskNumber - 1);
+                            String markMessage = taskToMark.markAsDone();
+                            System.out.println(markMessage);
+                        }
+                        userInput = sc.nextLine();
+                    } else if (userInput.startsWith("unmark")) {
+                        String[] strArr = userInput.split(" ", 2);
+                        int taskNumber = Integer.parseInt(strArr[1]);
+                        if (taskNumber < 1 || taskNumber > tl.items) {
+                            throw new InvalidTaskException(taskNumber);
+                        } else {
+                            Task taskToUnmark = tl.list.get(taskNumber - 1);
+                            String unmarkMessage = taskToUnmark.markAsUndone();
+                            System.out.println(unmarkMessage);
+                        }
+                        userInput = sc.nextLine();
+                    } else if (userInput.startsWith("todo")) {
+                        String[] strArr = userInput.split(" ", 2);
+                        String taskName = strArr[1];
+                        ToDo toDoTask = new ToDo(taskName);
+                        String addConfirmation = tl.addTask(toDoTask);
+                        System.out.println(addConfirmation);
+                        userInput = sc.nextLine();
+                    } else if (userInput.startsWith("deadline")) {
+                        String[] strArr = userInput.split(" ", 2);
+                        String[] strArr2 = strArr[1].split("/");
+                        if (strArr2.length != 2) {
+                            throw new IncorrectArgumentsException(userInput, 2, strArr2.length);
+                        }
+                        String taskName = strArr2[0];
+                        String dueTime = strArr2[1];
+                        Deadline deadlineTask = new Deadline(taskName, dueTime);
+                        String addConfirmation = tl.addTask(deadlineTask);
+                        System.out.println(addConfirmation);
+                        userInput = sc.nextLine();
```

Comment:
It seems like this code is repeated for each block. You may wish to abstract each block into classes so that it is neater and easier to read.